test/saithrift/tests/sail3.py

```diff
@@ -1477,3 +1477,207 @@ def runTest(self):
             print "END OF TEST"
+@group('lag')
+@group('l3')
+class L3MultipleEcmpLagTest(sai_base_test.ThriftInterfaceDataPlane):
+    # ports that will change from rif to lag member
+    total_changing_ports = 15
+    #the first port that will start as rif, that means that the first iteratio will only have port  #1
+    first_changing_port = 2
+    total_dst_port = 16
+    v4_enabled = 1
+    v6_enabled = 1
+    ip_mask = '255.255.0.0'
+    addr_family = SAI_IP_ADDR_FAMILY_IPV4
+    lag_members = []
+    nhop_group =  0
+    lag = 0
+    lag_rif = 0
+    port_rifs = []
+    neighbors = []
+    nhops = []
+    routes = []
+    vr_id = 0
+    mac_action = SAI_PACKET_ACTION_FORWARD
+    src_port = 0
+    mac_pool = ['00:11:22:33:44:50',
+               '00:11:23:33:44:51',
+               '00:11:24:33:44:52',
+               '00:11:25:33:44:53',
+               '00:11:26:33:44:54',
+               '00:11:27:33:44:55',
+               '00:11:28:33:44:56',
+               '00:11:29:33:44:57',
+               '00:11:30:33:44:58',
+               '00:11:31:33:44:59',
+               '00:11:32:33:44:60',
+               '00:11:33:33:44:61',
+               '00:11:34:33:44:62',
+               '00:11:35:33:44:63',
+               '00:11:36:33:44:64',
+               '00:11:37:33:44:65',
+               '00:11:38:33:44:66']
+    
+    
+    def setup_ecmp_lag_group(self, first_rif_port):
+        self.lag = self.client.sai_thrift_create_lag([])
+        #adding lag members
+        sai_thrift_create_lag_member(self.client, self.lag, port_list[1])
+        for i in range(self.first_changing_port,first_rif_port):
+            self.lag_members.append(sai_thrift_create_lag_member(self.client, self.lag, port_list[i]))
+        #adding router interfaces
+        self.lag_rif = sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.lag, 0, self.v4_enabled, self.v6_enabled, '')
+        for i in range(first_rif_port,self.total_changing_ports):
+            self.port_rifs.append(sai_thrift_create_router_interface(self.client, self.vr_id, 1, port_list[i], 0, self.v4_enabled, self.v6_enabled, ''))
+        #adding neighbors
+        sai_thrift_create_neighbor(self.client, self.addr_family, self.lag_rif, "10.10.16.1", self.mac_pool[15])
+        for i in range(self.total_changing_ports - first_rif_port):
+            sai_thrift_create_neighbor(self.client, self.addr_family, self.port_rifs[i], "10.10.%s.1" % str(i+1), self.mac_pool[i])
+            self.nhops.append(sai_thrift_create_nhop(self.client, self.addr_family, "10.10.%s.1" % str(i+1), self.port_rifs[i]))
+            sai_thrift_create_route(self.client, self.vr_id, self.addr_family, "10.10.%s.1" % str(i+1), '255.255.255.0', self.port_rifs[i])
+        sai_thrift_create_route(self.client, self.vr_id,self.addr_family, "10.10.16.1", '255.255.255.0', self.lag_rif)        
+        self.nhops.append(sai_thrift_create_nhop(self.client, self.addr_family, "10.10.16.1" , self.lag_rif))
```

Comment:
use 10.10.0.1 instead of 10.10.16.1 because if the total port is 17, it will have a conflict.